### PR TITLE
Update EIP-2537: Added MSM and pairings precompiles without subgroup checks

### PR DESCRIPTION
**Pull request by Marchhill**
- Removed redundant MUL operations to minimise increase in precompile count (as in #8945).
 - Added 'unsafe' MSM and pairings precompiles that do not have subgroup checks.
 - todo: Determine relative pricing of precompiles with and without subgroup checks

### CODE REVIEW

**Comment by asanso** [comment]
@Marchhill  may you motivate these proposed changes ?

**Comment by Marchhill** [comment]
> @Marchhill may you motivate these proposed changes ?

Added some motivation for unsafe precompiles.

File: EIPS/eip-2537.md [hunk 1]
```diff
@@ -24,28 +24,30 @@ The motivation of this precompile is to add a cryptographic primitive that allow
...
 - BLS12_G2ADD - to perform point addition in G2 (curve over quadratic extension of the base prime field) with a gas cost of `800` gas
-- BLS12_G2MUL - to perform point multiplication in G2 (curve over quadratic extension of the base prime field) with a gas cost of `45000` gas
-- BLS12_G2MSM - to perform multi-scalar-multiplication (MSM) in G2 (curve over quadratic extension of the base prime field) with a gas cost formula defined in the corresponding section
+- BLS12_G2MSM - to perform multi-scalar-multiplication (MSM) in G2 (curve over quadratic extension of the base prime field) with a gas cost formula defined in the corresponding section. In the base case of one point this will be a simple point multiplication.
+- BLS12_G2MSM_UNSAFE - identical to BLS12_G2MSM except that subgroup checks will **not** be performed.
 - BLS12_PAIRING_CHECK - to perform a pairing operations between a set of *pairs* of (G1, G2) points a gas cost formula defined in the corresponding section
+- BLS12_PAIRING_CHECK_UNSAFE - identical to BLS12_PAIRING_CHECK except that subgroup checks will **not** be performed.
```
Comment (chfast) [inline EIPS/eip-2537.md:50]: I think this is no-starter. How are you going to define this operation for points outside of the main prime subgroup?

**Review by chfast — CHANGES_REQUESTED** [review]

File: EIPS/eip-2537.md (hunk 1 above)
Comment (asanso) [reply EIPS/eip-2537.md:50]: totally agree with @chfast  here!!

**Pull request closed without merging**